columnphysics/icepack_therm_itd.F90

```diff
@@ -1021,215 +1015,142 @@ subroutine lateral_melt (dt,         ncat,       &
+                     end do
 
-                         ! fsd tendency
-                         do k = 1, nfsd
-                           d_afsd_tmp(k) = -df_flx(k) + c2 * G_radialn(n) * afsd_tmp(k) &
-                                       * (c1/floe_rad_c(k) - tmp)
-                         end do
+                     if (abs(sum(df_flx(:))) > puny) then
+                        write(warnstr,*) subname, 'sum(df_flx) /= 0'
+                        call icepack_warnings_add(warnstr)
+                     endif
 
-                         ! timestep required for this
-                         subdt = get_subdt_fsd(nfsd, afsd_tmp(:), d_afsd_tmp(:))
-                         subdt = MIN(subdt, dt)
+                     ! this term ensures area conservation
+                     tmp = SUM(afsd_tmp(:)/floe_rad_c(:))
 
-                        ! update fsd and elapsed time
-                        afsd_tmp(:) = afsd_tmp(:) + subdt*d_afsd_tmp(:)
-                        elapsed_t = elapsed_t + subdt
+                     ! fsd tendency
+                     do k = 1, nfsd
+                        d_afsd_tmp(k) = -df_flx(k) + c2 * G_radialn(n) * afsd_tmp(k) &
+                             * (c1/floe_rad_c(k) - tmp)
+                     end do
 
+                     ! timestep required for this
+                     subdt = get_subdt_fsd(nfsd, afsd_tmp(:), d_afsd_tmp(:))
+                     subdt = MIN(subdt, dt)
 
-                      END DO
+                     ! update fsd and elapsed time
+                     afsd_tmp(:) = afsd_tmp(:) + subdt*d_afsd_tmp(:)
+                     elapsed_t = elapsed_t + subdt
 
-                     afsdn(:,n) = afsd_tmp(:)
+                  END DO
 
+                  afsdn(:,n) = afsd_tmp(:)
 
-                  end if ! aicen
-               end if ! rside > 0, otherwise do nothing
+               end if ! aicen
+            end if ! rside > 0, otherwise do nothing
 
-            end if ! tr_fsd
+         end if ! tr_fsd
 
-            ! fluxes
-            do k = 1, nilyr
-               ! enthalpy tracers do not change (e/v constant)
-               ! heat flux to coupler for ice melt (dfhocn < 0)
-               dfhocn = trcrn(nt_qice+k-1,n)*rsiden(n) / dt &
-                      * vicen(n)/real(nilyr,kind=dbl_kind)
-               fhocn  = fhocn + dfhocn
-            enddo                  ! nilyr
-
-            do k = 1, nslyr
-               ! heat flux to coupler for snow melt (dfhocn < 0)
-               dfhocn = trcrn(nt_qsno+k-1,n)*rsiden(n) / dt &
-                      * vsnon(n)/real(nslyr,kind=dbl_kind)
-               fhocn  = fhocn + dfhocn
-            enddo                  ! nslyr
+         ! fluxes
+         do k = 1, nilyr
+            ! enthalpy tracers do not change (e/v constant)
+            ! heat flux to coupler for ice melt (dfhocn < 0)
+            dfhocn = trcrn(nt_qice+k-1,n)*rsiden(n) / dt &
+                 * vicen_init(n)/real(nilyr,kind=dbl_kind)
```

Comment:
Are the changes to _init values bug fixes?

Reply:
Yes.

Reply:
Are they the only bug fixes?  Please explain this in the main PR info (at the bottom, where it will be copied into the permanent record).